src/flows/components/header/TimeRangeDropdown.tsx

```diff
@@ -28,7 +29,10 @@ const TimeRangeDropdown: FC = () => {
     if (!flow.range) {
       return
     }
-    queryAll()
+    if (ref.current !== flow.range.lower) {
+      queryAll()
+      ref.current = flow.range.lower
+    }
```

Comment:
had to do this so we don't `queryAll` when this component is mounted (which happens when you toggle the share menu)

Reply:
Refs like this, similar to a "hasRun" boolean, are a red flag that something else is wrong.

Reply:
talked this out to myself on slack. the problem is that this component is in charge of running the query on change AND mount, which is causing issues when the component remounts as the header is toggled. instead of adding conditions to allow this component to manage that complexity, and since this useeffect has nothing to do with the render cycle, it makes sense to just remove the responsibility off of the view layer and closer to the data layer within the context.